test/CoreTest.cpp

```diff
@@ -567,6 +570,8 @@ TEST_CASE( "wrapper_test", "[core]" )
     REQUIRE(o.iterations[6].particles["electrons"].particlePatches["numParticles"][RecordComponent::SCALAR].m_chunks->size() == 1);
     REQUIRE(pp["numParticles"][RecordComponent::SCALAR].m_chunks->size() == 1);
 #endif
+    REQUIRE_THROWS_WITH(o.iterations[6].particles["electrons"].particlePatches["numParticles"][RecordComponent::SCALAR].store(idx+1, 42.),
+                        Catch::Equals("Datatypes of chunk data (DOUBLE) and dataset (UINT64) do not match."));
```

Comment:
Kudos for also including unit tests!

Reply:
sneakily improving coverage, one corner case at a time :)